### ISSUE TITLE
"source xxx not protocol value=1" is interpreted as "source not xxx protocol value=1"

### ISSUE THREAD

**Issue by luizluca** [firewalld/firewalld#842]
**What happened**:
I'm trying to silence log for traffic from a network scanner. I'm doing that adding a rich rule with high priority:

```
rule priority=32767 source ipset="my_scanners" not protocol value=icmp drop
```
I needed to exclude icmp because firewalld will accept it only after *_post chain. However, my surprise was when I listed nft rules:

ip saddr != @my_scanners meta l4proto icmp ct state { new, untracked } drop

That was not what I meant. Rich rules listing also changed what I added:

```
rule priority="32767" source NOT ipset="my_scanners" protocol value="icmp" drop
```

It moved the NOT to source. After some digging, it looks like protocol does not accept not.

**What you expected to happen**:
- syntax should avoid this confusion, not accepting NOT where it shouldn't be.
- protocol (and friends) should accept not.
- I would like something that could write the rule:
```
ip saddr @my_scanners meta l4proto != icmp ct state { new, untracked } drop
```

**How to reproduce it (as minimally and precisely as possible)**:
Add the cited rule, check the output

**Anything else we need to know?**:

**Environment**:
- Firewalld Version (if Fedora based `dnf info firewalld` or commit hash if developing from git `git log -n1 --format=format:"%H"`): firewalld-0.9.3-1.1.noarch
- Firewalld Backend (`cat /etc/firewalld/firewalld.conf | grep FirewallBackend`): nftables
- OS (e.g: `cat /etc/os-release`): OpenSUSE 15.3 
- Others:

**Comment by erig0** [comment]
Thanks for finding and reporting this bug.

> It moved the NOT to source. After some digging, it looks like protocol does not accept not.

You're right. It doesn't support `not`. Only `source` and `destination` do. So the rich rule parser should be rejecting the rich rule.

> **What you expected to happen**:
> 
>     * syntax should avoid this confusion, not accepting NOT where it shouldn't be.

Agree.

>     * protocol (and friends) should accept not.

Can you file a separate issue for this enhancement? One issue per feature: `protocol`, `port`, etc.

>     * I would like something that could write the rule:
> 
> 
> ```
> ip saddr @my_scanners meta l4proto != icmp ct state { new, untracked } drop
> ```

The above enhancements should get that for you.

**Comment by erig0** [comment]
You can mimic the `not` behavior with two rich rules.

Instead of using
```
rule priority=32767 source ipset="my_scanners" not protocol value=icmp drop
```
Use this:

```
rule priority=32767 source ipset="my_scanners" protocol value=icmp accept
rule priority=32766 source ipset="my_scanners" drop
```
